### PR TITLE
Feature/web socket

### LINKED ISSUE #1540: [envoy-gw] Add web socket support
### How will you implement it

---

### LINKED ISSUE #1550: Add web socket proxying support
### Describe your problem(s)
Add basic support for web socket proxying. 

### How will you implement it

---

### PR DESCRIPTION
**Pull request by NomadXD**
### Purpose
Adding basic web socket proxying support

### Issues

Fixes #1550

### Automation tests
 - Unit tests added: Yes
 - Integration tests added: No

### Tested environments

Ubuntu 20.04

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/pkg/api/file_read_utils.go [hunk 1]
```diff
@@ -21,36 +21,44 @@ package api
...
 // content.
 // The apictl project must be in zipped format. And all the extensions should be defined with in the openAPI
-// definition as only swagger.yaml is taken into consideration here.
+// definition as only swagger.yaml is taken into consideration here. For websocket APIs api.yaml is taken into
+// consideration. API type is decided by the type field in the api.yaml file.
 func ApplyAPIProject(payload []byte) error {
 	zipReader, err := zip.NewReader(bytes.NewReader(payload), int64(len(payload)))
 	var upstreamCerts []byte
 	newLineByteArray := []byte("\n")
+	var swaggwerJsn []byte
```
Comment (AmaliMatharaarachchi) [inline adapter/pkg/api/file_read_utils.go:58]:
```suggestion
	var swaggerJsn []byte
```

File: adapter/pkg/api/file_read_utils.go [hunk 2]
```diff
@@ -87,9 +95,37 @@ func ApplyAPIProject(payload []byte) error {
 			}
 			upstreamCerts = append(upstreamCerts, unzippedFileBytes...)
 			upstreamCerts = append(upstreamCerts, newLineByteArray...)
+		} else if strings.Contains(file.Name, apiDefinitionFilename) {
+			loggers.LoggerAPI.Debugf("fileName : %v", file.Name)
+			unzippedFileBytes, conversionErr := readZipFile(file)
+			if err != nil {
```
[the remark this answers is not in the document]
Comment (NomadXD) [reply adapter/pkg/api/file_read_utils.go:101]: yeah fixed it.

File: adapter/pkg/api/file_read_utils.go (hunk 1 above)
Comment (NomadXD) [reply adapter/pkg/api/file_read_utils.go]: fixed

File: adapter/pkg/api/file_read_utils.go [hunk 3]
```diff
@@ -87,9 +95,37 @@ func ApplyAPIProject(payload []byte) error {
...
+				loggers.LoggerAPI.Errorf("Error occured while reading the api definition file : %v %v", file.Name, err.Error())
```
Comment (AmaliMatharaarachchi) [inline adapter/pkg/api/file_read_utils.go:100]: Then, let's change it here as well.

Comment (NomadXD) [reply adapter/pkg/api/file_read_utils.go:100]: fixed it

**Review by VirajSalaka — APPROVED** [review]

**Pull request merged**